### PR TITLE
Adding input search fields and pagination to events table

### LINKED ISSUE #70: Allow user to search/filter events
**User statement**
*As a user, I would like to be able to search/filter any events that I may have, making them easier to find and categorize.*

**Proposed solution**
Make the HTML table searchable on different columns and values

**Development notes**
Add any other notes, context, or screenshots relevant to the ticket here

### PR DESCRIPTION
**Pull request by rudy-patel**
### What is it?
This PR adds search/filter ability along with pagination to the events table.

Closes #70.

### Screenshots
* General search
image

* Search by column
image

* Pagination
image

### CODE REVIEW

File: templates/events.html [hunk 1]
```diff
@@ -1,19 +1,19 @@
 {% extends "base.html" %}
 {% block content %}
-<div class="container">
+<div class="container mb-3 mt-3">
     <div class="row justify-content-center">
-        <h4>Recent Notifications</h4>
+        <h3>Events</h3>
```
Comment (rudy-patel) [inline templates/events.html:5]: Renaming to be consistent with the navbar.

File: templates/events.html [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
             <thead>
               <tr>
-                <th scope="col">Sensor</th>
-                <th scope="col">Type</th>
-                <th scope="col">Time</th>
-                <th scope="col">Action</th>
+                <th>Sensor</th>
+                <th>Type</th>
+                <th>Time</th>
+                <th>Action</th>
```
Comment (rudy-patel) [inline templates/events.html:13]: `col` tag wasn't needed

File: templates/events.html [hunk 3]
```diff
@@ -59,9 +59,130 @@ <h4>Recent Notifications</h4>
...
+                <td>Camera #1</td>
+                <td>Fault Detected</td>
+                <td>
+                    6:42 am
+                    27/02/2022
+                </td>
+                <td>
+                    <button type="button" class="btn btn-primary">Go to Train</button>
+                </td>
+              </tr>
```
Comment (rudy-patel) [inline templates/events.html:149]: You can ignore all this. Just duplicated the placeholder data that was already in the table to make sure the pagination was working.

File: templates/events.html [hunk 4]
```diff
@@ -59,9 +59,130 @@ <h4>Recent Notifications</h4>
...
             </tbody>
+
+            <tfoot>
+              <tr>
+                <th scope="col">Sensor</th>
+                <th scope="col">Type</th>
+                <th scope="col">Time</th>
+                <th scope="col">Action</th>
+              </tr>
+            </tfoot>
```
Comment (rudy-patel) [inline templates/events.html:159]: This footer tag inserts the footer at the bottom of the table, eventually making it searchable.

File: templates/events.html [hunk 5]
```diff
@@ -59,9 +59,130 @@ <h4>Recent Notifications</h4>
...
           </table>
-    </div>
+          <div class="d-flex justify-content-center mt-2">
+            <ul class="pagination"></ul>
+          </div>
```
Comment (rudy-patel) [inline templates/events.html:163]: Adding the pagination component.

File: templates/events.html [hunk 6]
```diff
@@ -59,9 +59,130 @@ <h4>Recent Notifications</h4>
...
 </div>
 
+<script>
+  
+  var table = $('.events-table').DataTable({
+    searching: true, // enable search input
+    lengthMenu: [[10, 25, 50, -1], [10, 25, 50, "All"]] // custom number of entries
+  });
```
Comment (rudy-patel) [inline templates/events.html:171]: Some javascript here, just setting the `searching` flag to be true and providing some length choices for the "results per page" dropdown menu.

File: templates/events.html [hunk 7]
```diff
@@ -59,9 +59,130 @@ <h4>Recent Notifications</h4>
...
+
+  table.columns().every( function () {
+    var that = this;
+    $('input', this.footer() ).on('keyup change', function () {
+      if ( that.search() !== this.value ) {
+        that.search( this.value ).draw();
+      }
+    });
+  })
```
Comment (rudy-patel) [inline templates/events.html:185]: This javascript code handles the search input from the footer of the table and updates the results in the table.

File: templates/home.html [hunk 8]
```diff
@@ -28,7 +28,7 @@ <h4>Camera #1 Active</h4>
         </div>
         <div class="col-4">
             <div class="row justify-content-center">
-                <h4>Recent Notifications</h4>
+                <h4>Recent Events</h4>
```
Comment (rudy-patel) [inline templates/home.html:31]: Again, just updating the wording to match the navbar.

**Review by kaden-dreger — APPROVED** [review]
Looks good

**Pull request merged**